source/base_interface.cc

```diff
@@ -397,6 +398,42 @@ output_solution (const unsigned int &current_cycle,
 
 }
 
+#ifdef DEAL_II_WITH_ARPACK
+
+template<int dim, int spacedim, typename LAC>
+void
+BaseInterface<dim,spacedim,LAC>::
+output_eigenvectors(const std::vector<typename LAC::VectorType> &eigenvectors,
+                    const std::vector<std::complex<double> > &eigenvalues,
+                    const unsigned int &current_cycle) const
+{
+  auto &pcout = this->get_pcout();
+
+  for (unsigned int i=0; i<eigenvectors.size(); ++i)
+    {
+      pcout << "eigenvalues[" <<i<<"] = " << eigenvalues[i]<< std::endl;
+      std::stringstream suffix;
+      suffix << ".eig." << current_cycle << "."<<i;
+      data_out.prepare_data_output(this->get_dof_handler(),suffix.str());
+
+      std::vector<std::string> eig_names =
+        Utilities::split_string_list(get_component_names());
+      for (auto &name : eig_names)
+        {
+          name += "_";
+          name += Utilities::int_to_string(i);
+        }
+
+      data_out.add_data_vector (eigenvectors[i],
+                                get_component_names());
+
+      data_out.write_data_and_clear(get_mapping());
```

Comment:
If I put everything in one file is (with paraview) more difficult to switch among eigenvectors. especially if I have more variables. This way the different eigenvectors are sees as different time step.

Tell me what do you prefear